src/Designer/frontend/dashboard/pages/OrgContentLibraryPage/utils.ts

```diff
@@ -27,38 +29,85 @@ export function textResourcesWithLanguageToLibraryTextResources({
 }
 
 export function backendCodeListsToLibraryCodeLists(
-  codeLists: CodeListsNewResponse,
+  response: GetSharedResourcesResponse | undefined,
 ): LibraryCodeListData[] {
-  return codeLists.codeListWrappers.filter(isWithoutError).map(({ codeList, title }) => ({
-    name: title,
-    codes: codeList.codes,
-  }));
+  if (!response) return [];
+
+  return response.files.map((file) => {
+    const fileName = file.path.split('/').pop()?.replace('.json', '') || 'unknown';
+
+    if (file.problem || !file.content) {
+      // TODO: We should show the user that a codelist is corrupted
+      return { name: fileName, codes: [] };
+    }
+
+    try {
+      const codeList = JSON.parse(atobUTF8(file.content));
+      return {
+        name: fileName,
+        codes: Array.isArray(codeList.codes) ? codeList.codes : [],
+      };
+    } catch {
+      // TODO: We should show the user that a codelist is corrupted
+      return { name: fileName, codes: [] };
+    }
+  });
 }
 
-function isWithoutError(codeListData: CodeListDataNew): codeListData is ValidCodeListData {
-  return codeListData.codeList && !codeListData.hasError;
+function atobUTF8(data: string) {
+  const decodedData = atob(data);
+  const utf8data = new Uint8Array(decodedData.length);
+  const decoder = new TextDecoder('utf-8');
+  for (let i = 0; i < decodedData.length; i++) {
+    utf8data[i] = decodedData.charCodeAt(i);
+  }
```

Comment:
Småpirk: Alternativ løsning som ikke innebærer `for`-løkke:
```suggestion
  [...decodedData].forEach((char, i) => utf8data[i] = char.charCodeAt(0));
```